xapian-bindings/java/java.i

```diff
@@ -174,8 +174,11 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next(int) "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
+    Xapian::docid next (int pseudoArg) {
```

Comment:
Is it necessary to add the check for `Xapian::iterator_valid(*self)` in next? In the foreach loop `for (MatchDescriptor descriptor: mset)`, `hasNext()` is called before every iteration and directly followed by a call to `next()`. It is equivalent to the pseudo code you suggested above, if we add a check to next then it would result in consecutive checks when we use the iterator.